src/cache_client.rs

```diff
@@ -552,6 +554,77 @@ impl CacheClient {
         request.send(self).await
     }
 
+    /// Check if the provided key exists in the cache
+    ///
+    /// # Arguments
+    /// * `cache_name` - name of cache
+    /// * `key` - key to check for existence
+    ///
+    /// # Examples
+    /// Assumes that a CacheClient named `cache_client` has been created and is available.
+    /// ```
+    /// # fn main() -> anyhow::Result<()> {
+    /// # use momento_test_util::create_doctest_cache_client;
+    /// # tokio_test::block_on(async {
+    /// # let (cache_client, cache_name) = create_doctest_cache_client();
+    /// use momento::requests::cache::scalar::key_exists::KeyExists;
+    ///
+    /// let result = cache_client.key_exists(&cache_name, "key").await?;
+    /// if result.exists {
+    ///     println!("Key exists!");
+    /// } else {
+    ///     println!("Key does not exist!");
+    /// }
+    /// # Ok(())
+    /// # })
+    /// # }
+    /// ```
+    /// You can also use the [send_request](CacheClient::send_request) method to get an item using a [KeyExistsRequest].
+    pub async fn key_exists(
+        &self,
+        cache_name: impl Into<String>,
+        key: impl IntoBytes,
+    ) -> MomentoResult<KeyExists> {
+        let request = KeyExistsRequest::new(cache_name, key);
+        request.send(self).await
+    }
+
+    /// Check if the provided keys exist in the cache.
+    /// Returns a list of booleans indicating whether each given key was found in the cache.
+    ///
+    /// # Arguments
+    /// * `cache_name` - name of cache
+    /// * `keys` - list of keys to check for existence
+    ///
+    /// # Examples
+    /// Assumes that a CacheClient named `cache_client` has been created and is available.
+    /// ```
+    /// # fn main() -> anyhow::Result<()> {
+    /// # use momento_test_util::create_doctest_cache_client;
+    /// # tokio_test::block_on(async {
+    /// # let (cache_client, cache_name) = create_doctest_cache_client();
+    /// use momento::requests::cache::scalar::keys_exist::KeysExist;
+    ///
+    /// let result = cache_client.keys_exist(&cache_name, vec!["key1", "key2", "key3"]).await?;
+    /// if !result.exists.is_empty() {
+    ///     println!("Processing list of booleans: {:#?}", result.exists);
```

Comment:
I suspect that as a user the main way I'd prefer to interact with this is to have basically a "HashMap" type of interface available on the result. So I can do like, `result.hash_map.get("key1")` or something like that. The list of booleans is necessary as a lowest-level fallback, but generally as a user I'd prefer to avoid it. Maybe worth a quick look to see if that would be easy to add? I think we have a few examples of this in other SDKs.

Reply:
Looks like it's inconsistent across SDKs. I agree that would be a good UX, can definitely add a hash map here and where they're currently missing:

**Provides hash map**
* [Dotnet sdk](https://github.com/momentohq/client-sdk-dotnet/blob/main/src/Momento.Sdk/Responses/CacheKeysExistResponse.cs)
* [PHP sdk](https://github.com/momentohq/client-sdk-php/blob/4fb88057f3a4cdad2f186c811043c171e1d10ce4/src/Cache/CacheOperationTypes/CacheOperationTypes.php#L862)

**Missing hash map**
* [JS sdk](https://github.com/momentohq/client-sdk-javascript/blob/main/packages/core/src/messages/responses/cache-keys-exist.ts)
* [Go sdk](https://github.com/momentohq/client-sdk-go/blob/main/responses/keys_exist.go)

Reply:
oh wow good catch. can you create a ticket for adding that to the JS SDK? :)

probably the dotnet sdk is a good one to use for inspiration for now though. feel free to timebox.